streaming/src/test/scala/org/apache/spark/streaming/scheduler/ReceiverTrackerSuite.scala

```diff
@@ -72,15 +77,50 @@ class ReceiverTrackerSuite extends TestSuiteBase {
     assert(locations(0).length === 1)
     assert(locations(3).length === 1)
   }
+
+  test("Receiver tracker - propagates rate limit") {
+    val newRateLimit = 100L
+    val ids = new TestReceiverInputDStream(ssc)
+    val tracker = new ReceiverTracker(ssc)
+    tracker.start()
+    eventually(timeout(5 seconds)) {
+      assert(TestDummyReceiver.started)
+    }
+    tracker.sendRateUpdate(ids.id, newRateLimit)
+    // this is an async message, we need to wait a bit for it to be processed
+    eventually(timeout(3 seconds)) {
+      assert(ids.getCurrentRateLimit.get === newRateLimit)
+    }
+  }
 }
 
+/** An input DStream with a hard-coded receiver that gives access to internals for testing. */
+private class TestReceiverInputDStream(@transient ssc_ : StreamingContext)
+  extends ReceiverInputDStream[Int](ssc_) {
+
+  override def getReceiver(): DummyReceiver = TestDummyReceiver
+
+  def getCurrentRateLimit: Option[Long] = {
+    TestDummyReceiver.executor.getCurrentRateLimit
+  }
+}
+
+/**
+ * We need the receiver to be an object, otherwise serialization will create another one
+ * and we won't be able to read its rate limit.
+ */
+private object TestDummyReceiver extends DummyReceiver
```

Comment:
Then let's call it SingletonReceiver. I find using test and dummy both in
the name very superfl

On Wednesday, July 22, 2015, Iulian Dragos <notifications@github.com
<javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:

> In
> streaming/src/test/scala/org/apache/spark/streaming/scheduler/ReceiverTrackerSuite.scala
> <https://github.com/apache/spark/pull/7471#discussion_r35208599>:
>
> > +/** An input DStream with a hard-coded receiver that gives access to internals for testing. */
> > +private class TestReceiverInputDStream(@transient ssc_ : StreamingContext)
> > +  extends ReceiverInputDStream[Int](ssc_) {
> > +
> > +  override def getReceiver(): DummyReceiver = TestDummyReceiver
> > +
> > +  def getCurrentRateLimit: Option[Long] = {
> > +    TestDummyReceiver.executor.getCurrentRateLimit
> > +  }
> > +}
> > +
> > +/**
> > + * We need the receiver to be an object, otherwise serialization will create another one
> > + * and we won't be able to read its rate limit.
> > + */
> > +private object TestDummyReceiver extends DummyReceiver
>
> Is your issue only about naming? I personally find it weird to have the
> companion object of a class extend the class itself. It's true that I
> *could* do that, but I prefer not to. If you feel strongly about it I can
> do the renaming you wish, but I find it unnecessary.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/7471/files#r35208599>.
>


-- 
Sent from Gmail Mobile